clinica/pipelines/t1_freesurfer_longitudinal/t1_freesurfer_longitudinal_correction_utils.py

```diff
@@ -269,28 +269,52 @@ def save_to_caps(source_dir, subject_id, caps_dir, overwrite_caps=False):
     return str_image_id
 
 
-def get_processed_images(caps_directory, part_ids, sess_ids, long_ids):
+def get_processed_images(caps_directory, part_ids, sess_ids, long_ids, atlas=False):
     """
     Extract image IDs (e.g. ["sub-CLNC01_ses-M00_long-M00M18", "sub-CLNC01_ses-M18_long-M00M18"]) of outputs
     already processed by T1FreeSurferLongitudinalCorrection pipeline.
     """
     import os
 
     image_ids = []
-    if os.path.isdir(caps_directory):
-        for (participant_id, session_id, long_id) in zip(part_ids, sess_ids, long_ids):
-            output_file = os.path.join(
-                os.path.expanduser(caps_directory),
-                "subjects",
-                participant_id,
-                session_id,
-                "t1",
-                long_id,
-                "freesurfer_longitudinal",
-                f"{participant_id}_{session_id}.long.{participant_id}_{long_id}",
-                "mri",
-                "aparc+aseg.mgz",
-            )
-            if os.path.isfile(output_file):
-                image_ids.append(f"{participant_id}_{session_id}_{long_id}")
+    if atlas:
+        if os.path.isdir(caps_directory):
+            for (participant_id, session_id, long_id) in zip(
+                part_ids, sess_ids, long_ids
+            ):
+                output_file = os.path.join(
+                    os.path.expanduser(caps_directory),
+                    "subjects",
+                    participant_id,
+                    session_id,
+                    "t1",
+                    long_id,
+                    "freesurfer_longitudinal",
+                    f"{participant_id}_{session_id}.long.{participant_id}_{long_id}",
+                    "stat",
+                    "rh.",
+                    atlas,
+                    ".stats",
+                )
+                if os.path.isfile(output_file):
+                    image_ids.append(f"{participant_id}_{session_id}_{long_id}")
+    else:
+        if os.path.isdir(caps_directory):
+            for (participant_id, session_id, long_id) in zip(
+                part_ids, sess_ids, long_ids
+            ):
+                output_file = os.path.join(
+                    os.path.expanduser(caps_directory),
+                    "subjects",
+                    participant_id,
+                    session_id,
+                    "t1",
+                    long_id,
+                    "freesurfer_longitudinal",
+                    f"{participant_id}_{session_id}.long.{participant_id}_{long_id}",
+                    "mri",
+                    "aparc+aseg.mgz",
+                )
+                if os.path.isfile(output_file):
+                    image_ids.append(f"{participant_id}_{session_id}_{long_id}")
```

Comment:
Looks like this can be refactored a bit. Looks to me that only the suffix changes, right?

Reply:
yup I can gain a few lines easily, you are right !